sparse_strips/vello_cpu/src/fine/mod.rs

```diff
@@ -507,20 +518,324 @@ pub(crate) mod strip {
+        Self::max(self, other)
+    }
+
+    #[inline(always)]
+    fn normalized_mul(self, other: Self) -> Self {
+        self * other
+    }
+
+    #[inline(always)]
+    fn narrow(self) -> Self {
+        self
+    }
+}
+
+impl Widened<u8> for u16 {
+    fn clamp(self) -> Self {
+        Ord::clamp(self, u8::ZERO as Self, u8::ONE as Self)
+    }
+
+    fn normalize(self) -> Self {
+        div_255(self)
+    }
+
+    fn min(self, other: Self) -> Self {
+        Ord::min(self, other)
+    }
+
+    fn max(self, other: Self) -> Self {
+        Ord::max(self, other)
+    }
+
+    fn normalized_mul(self, other: Self) -> Self {
+        (self * other).normalize()
+    }
+
+    fn narrow(self) -> u8 {
+        debug_assert!(
+            self <= u8::MAX as Self,
+            "cannot narrow integers larger than u8::MAX"
+        );
+
+        self as u8
+    }
+}
+
+/// A type that can be used as the underlying storage for fine rasterization.
+pub trait FineType:
+    Sized
+    + Copy
+    + PartialEq<Self>
+    + PartialOrd<Self>
+    + Add<Self, Output = Self>
+    + Mul<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + Debug
+{
+    type Widened: Widened<Self>;
+
+    /// The number that is considered to be the minimum of the normalized range of this type.
+    const ZERO: Self;
+    /// The number that is considered to be in the "center" of the normalized range of this type.
+    const MID: Self;
+    /// The number considered to be the maximum of the normalized range of the type.
+    const ONE: Self;
+
+    /// Return the minimum number.
+    fn min(self, other: Self) -> Self;
+    /// Return the maximum number.
+    fn max(self, other: Self) -> Self;
+    /// Perform a widening multiplication and then divide by a third number.
+    fn widened_mul_div(self, other: Self, other2: Self) -> Self::Widened;
+    /// Perform a normalized multiplication between this number and another
+    fn normalized_mul(self, other: Self) -> Self;
+    /// Extract the underlying color from a premultiplied color.
+    fn extract_color(color: &PremulColor) -> [Self; COLOR_COMPONENTS];
+    /// Convert a normalized u8 integer to this type.
+    fn from_normalized_u8(num: u8) -> Self;
+    /// Convert a plain u8 integer to this type.
+    fn from_u8(num: u8) -> Self;
+    /// Convert this number to a normalized f32.
+    fn to_f32_normalized(self) -> f32;
+    /// Convert to this number from a normalized f32.
+    fn from_normalized_f32(num: f32) -> Self;
+    // TODO: These should be sized to COLOR_COMPONENTS, but will leave that for
+    // the future.
+    /// Convert a slice to a RGBA8 slice.
+    fn to_rgba8(_in: &[Self]) -> [u8; COLOR_COMPONENTS];
+    /// Convert a RGBA8 slice to a slice of this type.
+    fn from_rgba8(_in: &[u8]) -> [Self; COLOR_COMPONENTS];
+    /// Convert a RGBAF32 slice to a slice of this type.
+    fn from_rgbaf32(_in: &[f32]) -> [Self; COLOR_COMPONENTS];
+    /// Get the inverse of the number.
+    fn inv(self) -> Self;
```

Comment:
The name "inverse" implies this is a multiplicative inverse. It is less common to speak about a "modular arithmetic inverse" (and it's also not exactly a modular arithmetic inverse, as `self + self.inv() == Self::ONE`, not `Self::ZERO`). The clearest name would probably be the following.

```suggestion
    /// Calculate "one minus" this number, i.e., `Self::ONE - self`.
    fn one_minus(self) -> Self;
```

Alternatively,

```suggestion
    /// Reflect `self` around the midpoint.
    fn reflect(self) -> Self;
```